packages/aws-rfdk/lib/deadline/lib/sep-spotfleet.ts

```diff
@@ -0,0 +1,731 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as crypto from 'crypto';
+import {
+  BlockDevice,
+  BlockDeviceVolume,
+  CfnLaunchConfiguration,
+  EbsDeviceVolumeType,
+} from '@aws-cdk/aws-autoscaling';
+import {
+  Connections,
+  IConnectable,
+  IMachineImage,
+  InstanceType,
+  ISecurityGroup,
+  IVpc,
+  OperatingSystemType,
+  Port,
+  SecurityGroup,
+  SubnetSelection,
+  UserData,
+} from '@aws-cdk/aws-ec2';
+import {
+  CfnInstanceProfile,
+  IGrantable,
+  IPrincipal,
+  IRole,
+  ManagedPolicy,
+  Role,
+  ServicePrincipal,
+} from '@aws-cdk/aws-iam';
+import {
+  Annotations,
+  CfnTag,
+  Construct,
+  Fn,
+  Expiration,
+  IResource,
+  Lazy,
+  Names,
+  ResourceEnvironment,
+  Stack,
+} from '@aws-cdk/core';
+import {
+  IScriptHost,
+  LogGroupFactoryProps,
+} from '../../core';
+import {
+  RFDK_VERSION,
+  tagConstruct,
+  TAG_NAME,
+} from '../../core/lib/runtime-info';
+import {
+  IRenderQueue,
+} from './render-queue';
+import {
+  IInstanceUserDataProvider,
+  WorkerInstanceConfiguration,
+} from './worker-configuration';
+
+/**
+ * The allocation strategy for the Spot Instances in your Spot Fleet
+ * determines how it fulfills your Spot Fleet request from the possible
+ * Spot Instance pools represented by its launch specifications.
+ */
+export enum SEPSpotFleetAllocationStrategy {
+  /**
+   * Spot Fleet launches instances from the Spot Instance pools with the lowest price.
+   */
+  LOWEST_PRICE = 'lowestPrice',
+  /**
+   * Spot Fleet launches instances from all the Spot Instance pools that you specify.
+   */
+  DIVERSIFIED = 'diversified',
+  /**
+   * Spot Fleet launches instances from Spot Instance pools with optimal capacity for the number of instances that are launching.
+   */
+  CAPACITY_OPTIMIZED = 'capacityOptimized',
+}
+
+/**
+ * Resource types that presently support tag on create.
+ */
+enum ISpotFleetResourceType {
+  /**
+   * EC2 Instances.
+   */
+  INSTANCE = 'instance',
+
+  /**
+   * EC2 Instances.
+   */
+  SPOT_FLEET_REQUEST = 'spot-fleet-request',
+}
+
+/**
+ * Properties for the Spot Event Plugin Worker Fleet.
+ */
+export interface SEPSpotFleetProps {
+  /**
+   * VPC to launch the worker fleet in.
+   */
+  readonly vpc: IVpc;
+
+  /**
+   * Endpoint for the RenderQueue, to which the worker fleet needs to be connected.
+   */
+  readonly renderQueue: IRenderQueue;
+
+  /**
+   * AMI of the deadline worker to launch.
+   */
+  readonly workerMachineImage: IMachineImage;
+
+  /**
+   * The number of units to request for the Spot Fleet.
+   *
+   */
+  readonly targetCapacity: number;
+
+  /**
+   * Types of instances to launch.
+   */
+  readonly instanceTypes: InstanceType[];
+
+  /**
+   * Deadline groups these workers needs to be assigned to. The group is
+   * created if it does not already exist.
+   *
+   * @default - Workers are not assigned to any group
+   */
+  readonly deadlineGroups: string[];
```

Comment:
Unfortunately, `groups` and `pools` are not created by default, unlike we do in `WorkerFleet`. This is because to create a worker fleet request, we first need to:
* have an existing group
* have an existing pool
* submit a job with the above group and pool

So now these groups and pools have to be created manually (actually, some of them can be created, but still only after the first worker starts).

Reply:
I've already removed this from the TSDoc:
> The group is created if it does not already exist.

I also added to the README of ALL-In-AWS-Infrastructure-SEP a note, that customers have to create groups, pools and submit a job and only then the Spot Fleet Request will be crated. I will add the same information to the documentation of the constructs.

Reply:
Sounds great. This in combination with similar guidance in `packages/aws-rfdk/lib/deadline/README.md` should get the message across. Thanks!